src/com/palmergames/bukkit/towny/command/TownyAdminCommand.java

```diff
@@ -1268,4 +1279,250 @@ public void parseToggleCommand(String[] split) throws TownyException {
 		}
 	}
 
+	public static void handleTownMetaCommand(Player player, String[] split) throws TownyException {
+		Town town = null;
+		TownyUniverse townyUniverse = TownyUniverse.getInstance();
+
+		try {
+			town = townyUniverse.getDataSource().getResident(player.getName()).getTown();
+		} catch (Exception e) {
+			TownyMessaging.sendErrorMsg(player, e.getClass().toString());
+			return;
+		}
+
+		if (!townyUniverse.getPermissionSource().testPermission(player, PermissionNodes.TOWNY_COMMAND_TOWNYADMIN_TOWN_META.getNode()))
+			throw new TownyException(TownySettings.getLangString("msg_err_command_disable"));
+
+		if (split.length == 2) {
+			if (town.hasMeta()) {
+				player.sendMessage(ChatTools.formatTitle("Custom Meta Data"));
+				for (CustomDataField field : town.getMetadata()) {
+					player.sendMessage(field.getKey() + " = " + field.getValue());
+				}
+			} else {
+				TownyMessaging.sendErrorMsg(player, "This town doesn't have any associated metadata");
+			}
+
+			return;
+		}
+
+		if (split.length < 4) {
+			player.sendMessage(ChatTools.formatCommand("", "meta", "set", "The key of a registered data field"));
+			player.sendMessage(ChatTools.formatCommand("", "meta", "add", "Add a key of a registered data field"));

```

Comment:
needs to have the remove command added to help output

player.sendMessage(ChatTools.formatCommand("", "meta", "remove", "Add a key of a registered data field"));